# 인사

안녕하세요 ~~ 25기 여러분 반갑습니다 ~~  

드디어 정규세션이 시작되었네요.  

뭘 과제로 내볼까 고민하다가, 요즘 데이터 분석 채용 공고 자격 요건에 필수로 들어있는.... A/B TEST가 좋겠다!!고 생각했습니다.  

데이터 분석쪽이 아니더라도, 다양한 도메인에서 쓸 수 있는 방법이라고 생각해요.


A/B TEST의 개념부터 기초적인 과정을 이해해볼 수 있게 내봤어요. 근데 설명해야할 게 많다보니 텍스트가 너무 많아졌네요.

읽기 힘들어도 ㅎㅎ... 열심히 해주시고, 질문 사항은 저에게 연락 남겨주세요.  

참고로, 정규세션 발표 외의 내용을 구글링하고 공부해야만 풀 수 있는 문제들이 포함되어 있습니다.

제 이름... 아시죠...? 24기 박현서입니다....



<br>

# Q1. A/B TEST란 무엇일까요?   
또한, A/B TEST시 유의할 사항들을 초두효과를 포함해 기술해주세요.

정규세션에서 발표한 내용 외에 추가적으로 공부한 개념을 작성하셔도 됩니다.
정의와 함께 예시 하나만 들어주세요 ~~

## A1. 답안 작성  
답변: A/B Test란, 두 가지 버전(A, B)의 성과를 클릭률이나 전환률과 같은 평가 기준을 기반으로 각 버전의 성과를 비교하는 실험을 의미한다. 이 과정에서 테스트를 위한 각 버전을 설계할 때, 추론 통계적 특성을 활용해 실험을 설계하고 해당 가설을 검정하게 된다.<br>

A/B 테스트 시에 유의할 점들이 여러 가지 있는데, 첫 번째는 **초두효과(Primacy Effect)** 이다. 초두효과는 사용자가 기존 서비스 방식과 다른 서비스를 접했을 때 이에 적응하기 위한 시간이 필요함을 의미한다. A/B 테스트를 위해 처음 수정한 UI에 대하여 사용자가 적응하지 못했다면, 사용자의 혼란으로 인해 초기에는 일시적으로 이용률이 하락하는 등의 부정적인 반응이 있을 수 있다.<br>
초두효과와 반대되는 개념으로 **신기효과(Novelty Effect)** 가 있다. 이는 초두효과와 달리 새로 바뀐 방식에 대하여 사용자들이 호기심을 가져 일시적으로 높은 관심을 보이는 것을 의미한다. A/B test에서 처음 적용된 UI에 사용자들이 관심을 갖고 긍정적인 성과가 측정될 수 있지만, 추후에는 이전과 같거나 더 낮은 수준으로 수렴하는 경우가 있을 수 있다.<br>
위 초두효과와 신기효과는 A/B 테스트로 증명하고자 하는 가설의 실질적인 영향보다는 단기적인 왜곡이 발생할 수 있으므로 이 두 효과에 유의하여 데이터가 안정화 될 때까지 기다리는 과정을 거친 후에 테스트 결과를 해석하고 분석할 필요가 있다.<br>

또한, 테스트 시에 충분한 통계 집단을 대상으로 한 테스트인지를 확인해야 한다. 테스트한 집단의 수가 충분하지 못하다면, 이는 A/B 테스트 결과에 대한 충분한 근거가 되지 못할 수 있기 때문이다. 만일 테스트에 참여한 사용자가 5명 뿐이라면, 이렇게 추정된 결과는 사용자가 100명인 테스트에 비하여 테스트의 결과를 직접적으로 반영한다고 보기 어렵다. 띠라서 신뢰할 수 있는 p-value를 얻고자 적절한 크기의 표본이 필요하다.<br>
마지막으로 테스트 시의 조건은 동일하게 설계해야 한다. 테스트하고자하는 패턴 이외의 UI에 차이가 있거나, 개선이 이루어진다면 이는 순수하게 A/B 테스트로 증명하고자 하는 성과만을 반영했다고 보기 어렵다.



---



 A/B TEST를 하기 위해서는 1-pager를 작성해야합니다.  

1-pager란 문제 정의, 관찰하고자하는 실험대상, 가설설정, 평가지표 등을 작성하는  
간단히 말해 계획서를 의미하는데요,

다른 부서와의 소통을 위한 역할 또,과거에 어떤 테스트를
어떻게 진행했는지에 대한 기록의 역할을 하기도 합니다.


 A/B TEST의 1-pager는 복잡한 실험 과정을 한눈에 파악하고, 의사결정권자가 "그래서 이 실험을 왜 했고, 결과가 무엇이며, 무엇을 할 것인가?"를 바로 이해할 수 있도록 작성하는 것이 핵심입니다.

예시를 통해 어떤 항목이 있고, 각 항목에 어떤 내용을 채우는 건지 보여드릴게요.



---



**예시**   
(푸는 거 아니고, 예시입니다! 읽으면서 이런 부분은 이런 내용을 채우는 거군... 이해하시면 돼요!)  




 📊 [ThinkGood] 메인 홈 검색 모듈 상위 노출 실험 1-Pager

---

**1. 실험 전략 및 배경 (Strategy & Observation)**

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 특정 공모전을 찾으려는 목적이 뚜렷하지만, 현재 하단에 위치한 검색바를 즉각적으로 발견하지 못한 유저 |
| **Observation** | **[데이터 분석 결과]** <br> • 검색 이용 유저의 상세 진입 전환율(CVR)은 40%, 미이용 유저의 전환율은 25%임 <br> • 현재 검색 이용률은 30%로, 검색 가시성 개선을 통한 전환율 상승 여력이 큼 |
| **Problem Statement** | 검색바가 배너 하단에 위치하여 시인성이 낮으며, 유저가 정보를 탐색하기 전 이탈할 확률이 높음 |
| **Hypothesis** | 하단 검색바를 페이지 **최상단으로 노출**시켜 검색 접근성을 높인다면, 전체 공모전 상세 진입률(CVR)이 개선될 것이다 |

---

 **2. 실험군 설계 (Experimental Groups)**

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="https://i.postimg.cc/kgcLc7Qy/Gemini-Generated-Image-enqvuhenqvuhenqv-(1).png" width="100%"><br>
      <sub>검색바 하단 위치 (기존 UI)</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="https://i.postimg.cc/0N51Rz5M/Gemini-Generated-Image-enqvuhenqvuhenqv.png" width="100%"><br>
      <sub>검색바 최상단 배치 (변경 UI)</sub>
    </td>
  </tr>
</table>

---

 **3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** 공모전 상세 페이지 진입률 (Detail Page CVR)
> * **Secondary Metric (보조지표):** 검색바 클릭률 (Search CTR)
> * **Guardrail Metric (방어지표):** 메인 페이지 이탈률 (Bounce Rate)

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 유의수준 0.05 |
| **Statistical Power ($1-\beta$)** | 80% | 검정력 유지 |
| **Minimum Sample Size** | **총 92,000명** | 각 그룹당 46,000명 할당 |
| **Required Period** | **최소 4일** |  

---

**4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off:**   
    검색바 상단 노출로 인해 메인 배너의 시각적 주목도가 분산될 수 있으며, 기존 UI에 익숙한 유저에게 일시적인 초두 효과(Primacy Effect)에 의한 혼란이 발생할 수 있음.
* **Andon (중단 기준):**
    1. 대조군 대비 전체 상세 진입률이 **5% 이상 유의미하게 하락**할 경우
    2. UI 변경으로 인한 페이지 오류나 유저 이탈률이 **3일 연속 급증**할 경우

---







<br>

# Q2. 1-pager를 작성해봅시다. (꼬리문제 3 + 보고서 채우는 것1)
저희가 만든 데모 사이트 링크를 첨부해드릴테니, 마음껏 보신 뒤에 위의 예시처럼 1-pager를 작성해보기 바랍니다.

여기서 실험군 설계에 들어가는 사진은 굳이 넣을 필욘 없고 말로만 설명해주셔도 됩니다. 아니면 gemini에게 요청해볼 수도 있겠네요!  

또한 문제탐색에 필요한 cvr과 같은 지표들은 직접 가정하고 하시면 됩니다. 다만, 해당 지표들이 큰 문제가 있는 지표여야... a/b test를 진행하는 이유가 되겠죠!?


정리해보면, 저희 웹사이트 보시고 '이건 여기에 배치하면 더 좋을 거 같은데?' 하는 것들을 찾아서 모든 것을 가정한 뒤에 쓰시면 됩니다!

[데모 웹사이트](https://huggingface.co/spaces/HanJaeHui/ThinkGood_Recommendation_System)


추가적으로 아래 실험군 항목에 사진을 넣으실 분들은 제가 아래 첨부드리는 사이트에 방문하셔서 해당 이미지를 코랩에 넣을 수 있는 형태로 변환한 링크를 복붙하시면 됩니다.  

(.png로 끝나는 링크 복붙하시면 됩니다.)

[포스트 이미지 사이트](https://postimages.org/)



<br>

## Q2-1. 실험군을 설계 해봅시다.
아래 보고서를 채워 주세요.

## A2-1. 아래 보고서 항목 설명을 참고하여 '[ ]' 부분을 지우고 채워주세요.  
  



📊 [프로젝트명] 실험 기획안 1-Pager

---

1. 실험 전략 및 배경 (Strategy & Observation)

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 특정 공모전에 대한 목적은 뚜렷하지 않으나, 공모전을 리서칭하고자 들어온 이용자 |
| **Observation** | **데이터 근거** <br> • 검색 이용 유저의 상세 진입 전환율(CVR)은 40%, 미이용 유저의 전환율은 25%(위의 1-Pager 예시에 나와있는 부분) <br> • 페이지 체류 시간을 더 늘리고, 더 나아가 검색 미이용 유저의 상세 진입 전환률을 높이고자 함 |
| **Problem Statement** | 현재는 로그인 유저에 대해 개인화된 추천만을 제공하기 때문에<br>보편적으로 인기 있는 공모전을 둘러보고자 하는 유저의 니즈를 파악할 수 없음 |
| **Hypothesis** | **로그인된 계정에서도 인기순 정렬 리스트를 제공한다면, 검색 미이용 유저의 상세 진입 전환률이 높아질 것이다.**  |

---

 2. 실험군 설계 (Experimental Groups)

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="https://i.postimg.cc/13GDc8Nf/image.png" width="100%"><br>
      <sub>[오늘의 맞춤 추천, 놓치면 아쉬운 발견 등 종합 적합도를 기반으로 한 추천만을 제공]</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="https://i.postimg.cc/qqGz63Fd/image.png" width="100%"><br>
      <sub>[인기 공모전을 가장 상단에 배치하고, 그 밑에 개인화된 추천 공모전 리스트를 제공]</sub>
    </td>
  </tr>
</table>


<br>

## Q2-2. 실험에 필요한 최소 test users 수를 구해봅시다.
유의수준 0.05, 검정력 80% 하에 수행해주세요.  


아래 공식 참고해서, 최소 test users 수를 구하는 코드 작성해주세요.

추가적으로 n은 산출하는데에 가정한 p1, p2도 같이 작성해주세요.

$$n = \frac{(Z_{\alpha/2} + Z_{\beta})^2 \cdot [p_1(1-p_1) + p_2(1-p_2)]}{(p_1 - p_2)^2}$$

## A2-2. 아래 질문에 대한 답과 코드를 채워주세요.

p1, p2는 각각 무엇을 의미하는 확률이고, 어떤 값으로 가정하셨나요?   
또한 위 공식을 보고 n을 구하기 위한 코드를 작성해주세요. 단, 라이브러리는 z값을 구하는 용도의 것만 사용해주세요.


### 답변:
p1은 검색 미이용자의 기존 상세 진입 전환률인 25%로 지정했고, 30% 이상의 전환률을 목표로 하고잦 하므로 p2는 30%로 지정하였다.

In [2]:
# A2-2 코드작성. 나온 n은 아래 Minimum Sample Size에 기재.

import math
from scipy.stats import norm

alpha = 0.05
beta = 0.8
p1 = 0.25  # 기존 전환률
p2 = 0.3   # 목표 전환률

z_alpha = norm.ppf(1-alpha/2)
z_beta = norm.ppf(beta)


### 공식 계산 부분 ###
numerator = (z_alpha + z_beta)**2 * (p1 * (1 - p1) + p2 * (1 - p2))
denominator = (p1 - p2)**2

n = numerator/denominator

print(f"n = {n}")


#########

n = 1247.9718777615055


<br>

## Q2-3. 실험에 필요한 최소 실험 기간을 구해봅시다.

이 실험에 필요한 최소 test users가 92000명이 나왔고  
해당 웹페이지 하루 평균 방문자수가 50000명 정도라고 할 때 우리는 최소 4일이 필요합니다.

비교하고 싶은 대상이 총 두개이고, 하나의 대상을 실험하기 위해선 최소 이틀은 필요하기 때문이죠!

위에서 구하신 test users수를 바탕으로, 하루 평균 방문자수를 가정한 뒤 기간을 설정해주세요.  
구하신 기간은 밑에 표에 기재해주시면 됩니다.



## A2-3. 아래 세질문에 답하신 뒤 아래 보고서를 채워주세요.

가정한 하루 평균 방문자 수?  400명

산출된 최소 test users 수?   1248명(반올림)
  
그래서 필요한 최소 기간은?   8일

## **아래 보고서도 채워주세요!**  
**'[ ]' 부분을 지우고 해당 내용에 부합하는 내용들을 채워주시면 됩니다.**

---

**3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** 공모전 상세 페이지 진입률 (Detail Page CVR)
> * **Secondary Metric (보조지표):** 페이지 체류 시간
> * **Guardrail Metric (방어지표):** 메인 페이지 이탈률 (Bounce Rate)

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 보통 5%(0.05) 설정 |
| **Statistical Power ($1-\beta$)** | 80% | 보통 80% 설정 |
| **Minimum Sample Size** | **총 1248명** | 각 그룹당 628명 할당 |
| **Required Period** | **최소 8일** | 실험 진행 예정 기간 |

---

 **4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off :**    
    개인화된 추천이 아래로 내려가면서 기존 UI에 익숙했던 사용자로부터의 초두효과에 의한 혼란이 발생할 수 있음. 개인화된 추천을 기대한 사용자의 이탈이 발생할 수 있음.

* **Andon (중단 기준) :**
    1. 대조군 대비 전체 상세 진입률이 5% 이상 유의미하게 하락할 경우
    2. UI 변경으로 인한 페이지 오류나 유저 이탈률이 3일 연속 급증할 경우

---

<br>

# Q3. A/B TEST를 수행해봅시다.

1-pager를 작성했으면 직접 test도 수행해봐야겠죠!  
위에 제가 만들었던 1-pager를 바탕으로 검정을 해볼게요.  
대표적인 검정방법에는 t-test가 있고 카이제곱 겁정이 있습니다.  


---



[실험 가설]  
귀무가설 ($H_0$): 메인 홈 검색 모듈의 위치(하단 vs 최상단)와 상세 페이지 진입 여부 사이에는 상관관계가 없다. 즉, 두 그룹의 상세 페이지 진입률($CVR$)은 동일하다.$$H_0: p_A = p_B$$대립가설 ($H_1$): 메인 홈 검색 모듈을 최상단에 노출하는 것은 상세 페이지 진입 여부에 영향을 미친다. 즉, 실험군(B)의 상세 페이지 진입률이 대조군(A)보다 높을 것이다.$$H_1: p_A < p_B$$


<br>

## Q3-1. 위 가설에 적합한 검정 이름과 왜 해당 검정을 써야하는지 이유도 작성해주세요. (t test와 카이제곱검정 중)

## A3-1.  검정 이름과 이유를 설명해주세요.

답변: 위 가설에 필요한 검정은 카이제곱검정 이고,  

이유는 위 가설은 범주형 데이터인 CVR을 기준으로 하는 가설이기 때문이다.

<br>

In [ ]:
'''''''''''''''''''''''''''''''''''''''
제가 세운 A/B TEST 설계 실험 코드입니다.
'''''''''''''''''''''''''''''''''''''''

import pandas as pd
from scipy.stats import chi2_contingency

# 1. 관측 데이터 입력 (실제 실험 결과 수치로 변경하세요)
# [상세진입 성공 수, 상세진입 실패 수]
control_results = [11500, 34500]  # 대조군(A)
variant_results = [12190, 33810]  # 실험군(B)

# 2. 분할표(Contingency Table) 생성
data = [control_results, variant_results]

# 3. 카이제곱 검정 수행
chi2, p_value, dof, expected = chi2_contingency(data)

# 4. 결과 출력
print(f"카이제곱 통계량: {chi2:.4f}")
print(f"P-value: {p_value:.10f}")

alpha = 0.05
if p_value < alpha:
    print(f"결과: P-value가 {alpha}보다 작으므로 귀무가설을 기각합니다.")
    print("=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.")
else:
    print(f"결과: P-value가 {alpha}보다 크므로 귀무가설을 기각할 수 없습니다.")
    print("=> 검색바 위치 변경의 효과가 통계적으로 증명되지 않았습니다.")

카이제곱 통계량: 26.9884
P-value: 0.0000002047
결과: P-value가 0.05보다 작으므로 귀무가설을 기각합니다.
=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.


실제 데이터가 없어서, 저도 필요한 데이터는 가정했답니다.  

이제 25기 분들의 A/B TEST를 수행해보시죠!

<br>

## Q3-2. Q2에서 작성한 실험 수행을 위해 가설을 설정해주세요.


## A3-2. 아래 두 가설 내용을 채워주세요.
  
[실험 가설]  
귀무가설 ($H_0$): 로그인 유저의 인기순 추천 제공은 상세 페이지 진입 여부 사이에 상관관계가 없다. 즉, 두 그룹의 상세 페이지 진입률($CVR$)은 동일하다.
$$H_0: p_A = p_B$$
  
대립가설 ($H_1$): 로그인 유저의 인기순 추천 제공은 상세 페이지 진입 여부에 영향을 미친다. 즉, 실험군(B)의 상세 페이지 진입률이 대조군(A)보다 높을 것이다.
$$H_1: p_A < p_B$$


<br>

## Q3-3. 위 가설을 수행하기 위해 어떤 검정 방법을 사용해야하고 이유는 무엇인지 작성해주세요.

## A3-3. 아래 검정 이름과 이유를 채워주세요.
답변: 위 가설에 필요한 검정은 카이제곱검정 이고,  

이유는 위 가설 또한 범주형 데이터인 CVR을 기준으로 하는 가설이기 때문이다.

<br>

## Q3-4. 위에서 답한 것을 토대로, 검정을 수행해주세요. (t test와 카이제곱 검정 중)


## A3-4 아래 셀에 코드 작성해주세요. 그리고!!!
그리고 나온 값을 통해 수행하신 a/b test 결과를 설정하신 가설에 맞게 해석해주세요.

In [3]:
### A3-4. 코드작성

import pandas as pd
from scipy.stats import chi2_contingency

# 1. 관측 데이터 입력 (실제 실험 결과 수치로 변경하세요)
# [상세진입 성공 수, 상세진입 실패 수]
control_results = [152, 476]  # 대조군(A)
variant_results = [195, 433]  # 실험군(B)

# 2. 분할표(Contingency Table) 생성
data = [control_results, variant_results]

# 3. 카이제곱 검정 수행
chi2, p_value, dof, expected = chi2_contingency(data)

# 4. 결과 출력
print(f"카이제곱 통계량: {chi2:.4f}")
print(f"P-value: {p_value:.10f}")

alpha = 0.05
if p_value < alpha:
    print(f"결과: P-value가 {alpha}보다 작으므로 귀무가설을 기각합니다.")
    print("=> 로그인 유저의 인기순 추천 제공이 상세 페이지 진입률에 유의미한 변화를 주었습니다.")
else:
    print(f"결과: P-value가 {alpha}보다 크므로 귀무가설을 기각할 수 없습니다.")
    print("=> 로그인 유저의 인기순 추천 제공의 효과가 통계적으로 증명되지 않았습니다.")




#############

카이제곱 통계량: 7.0242
P-value: 0.0080416847
결과: P-value가 0.05보다 작으므로 귀무가설을 기각합니다.
=> 로그인 유저의 인기순 추천 제공이 상세 페이지 진입률에 유의미한 변화를 주었습니다.


**결과해석**

답변: P-value는 이 결과의 신뢰도로 이해할 수 있으며, 실험의 신뢰도가 기존에 기준으로 잡았던 0.05보다 작은 0.008 정도로 결과가 나왔으므로 이는 실험군(B)의 성과가 우연이 아니며 신뢰할 수 있다고 해석할 수 있다.<br>
이에 따라 탐색형 유저에 대한 니즈를 개인화된 추천 뿐 아니라 인기순 추천을 통해서도 충족시킬 수 있으며, 탐색형 유저에 대한 UI적인 개선을 적극적으로 도입해야 할 것으로 보인다.

<br>

**1주차 기초통계 과제는 여기까지였습니다. 수고하셨습니다!**